craft_application/application.py

```diff
@@ -248,10 +247,9 @@ def _get_dispatcher(self) -> craft_cli.Dispatcher:
         # Set the logging level to DEBUG for all craft-libraries. This is OK even if
         # the specific application doesn't use a specific library, the call does not
         # import the package.
-        craft_libs = ["craft_archives", "craft_parts", "craft_providers", "craft_store"]
-        for craft_lib in craft_libs:
-            logger = logging.getLogger(craft_lib)
-            logger.setLevel(logging.DEBUG)
+        util.setup_loggers(
```

Comment:
I wonder if it would be simpler (for the downstream applications) to have an overrideable `app.get_libs_to_log()` (name tbd)? It could be used in addition to this new utility function.
My reasoning is just selfish ergonomics: If, say, rockcraft needs to log an additional lib, I'd need to figure out a place to call `util.setup_loggers("my_new_lib")`, which isn't immediately obvious if I otherwise don't override `_get_dispatcher()`.

Reply:
I considered that initially but decided to go for this because we're moving an awful lot into the Application. The "blessed" place to put it would be in your `main()` function before calling `app.run()` (and perhaps before even instantiating the application).

I don't really have strong feelings either way though.

Reply:
OK, I've pushed an alternative that I think might satisfy us both. It prevents "yet another method" on the Application while making a PEP 20 compliant One Obvious Way to Do It™.